src/cli.c

```diff
@@ -0,0 +1,115 @@
+#include "cli.h"
+#include "logging.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <limits.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <errno.h>
+
+/******************************************************
+ *** LOCAL DEFINES                                  ***
+ ******************************************************/
+
+#define MAX_HZ (1000000L)
+
+/******************************************************
+ *** LOCAL VARIABLES                                ***
+ ******************************************************/
+
+static const char* usage_str = "yobemag [-d <0..4>] <ROM>";
+
+/******************************************************
+ *** LOCAL METHODS                                  ***
+ ******************************************************/
+
+static int safe_strtol(const char* str_to_conv, int* store_into) {
+    char* end;
+    const long strtol_in = strtol(str_to_conv, &end, 10);
+
+    errno = 0;
+
+    if (end == str_to_conv) {
+        LOG_FATAL("%s: not a decimal number", str_to_conv);
+    } else if ('\0' != *end) {
+        LOG_FATAL("%s: extra characters at end of input: %s", str_to_conv, end);
+    } else if ((LONG_MIN == strtol_in || LONG_MAX == strtol_in) && ERANGE == errno) {
+        LOG_FATAL("%s out of range of type long", str_to_conv);
+    } else if (strtol_in > INT_MAX) {
+        LOG_FATAL("%ld greater than INT_MAX", strtol_in);
+    } else if (strtol_in < INT_MIN) {
+        LOG_FATAL("%ld less than INT_MIN", strtol_in);
+    } else {
+        *store_into = (int)strtol_in;
+        return EXIT_SUCCESS;
+    }
+
+    return EXIT_FAILURE;
+}
+
+/******************************************************
+ *** EXPOSED METHODS                                ***
+ ******************************************************/
+
+CLIArguments* cli_config_default(void) {
+    CLIArguments* conf = malloc(sizeof(CLIArguments));
+
+    conf->debug = FATAL;
+    conf->us_delay = 0;
+    conf->rom_path = "";
+
+    return conf;
+}
+
+void cli_config_destroy(CLIArguments *conf) {
+    free(conf);
+}
+
+int cli_config_handle(CLIArguments* const conf, int argc, char **argv) {
+    int c;
+
+    if (argc < 2) {
+        LOG_FATAL("No ROM path specified!");
+        fprintf(stderr, "%s\n", usage_str);
+        return EXIT_FAILURE;
+    }
```

Comment:
Is this LOG correct? I mean we could do something like `./yobemag <rom_path>`

Also shoudn't we have a function that takes care of something like this (similar to strerror from errno.h), since this happens quite often:

```C
const char* get_err_str(ErrorCode err_code) {
    switch (err_code) {
        case TOO_FEW_ARGS:
            return "Too few arguments: " + usage_str;
         case ...
}
```

Then this if could be:
```C
if (argc < 2) {
   LOG_FATAL(get_err_code(TOO_FEW_ARGS));
   return EXIT_FAILURE;
}
```

Another problem I see here is the spamming of the stdout/err.
We are printing something along the lines "too few args: usage not correct", but also propagate the error back to the caller. Where, in this case, "Failed to load ROM" is printed. At that point, it's already clear that something went wrong. We don't need the second message. Even if we left the "failed to load ROM" message, the reverse order of the messages would make more sense.

This is one of the problems I see with error propagation 🤔 

An alternative would be (not using propagation):

```C
if (argc < 2) {
    YOBEMAG_EXIT(TOO_FEW_ARGS);
}
```

This could do the following things:
- print a message like: `FATAL - time - cli.c:74 - too few args: usage ...`
- clean up memory
- flush stderr and stdout
- exit(EXIT_FAILURE)

A problem with this approach is that cleaning up the memory could be non-trivial. However, as I see it, it shouldn't be in our scenario because we do very few allocation or could even use the stack for 8 KB of memory (which would even be faster).

Reply:
Generally, `argc` includes the path to the binary as well, so if `argc < 2`, only the path to the binary was specified.

Otherwise, that's a good idea. Would you mind to open an issue for that?

Reply:
Oh I see 🤦🏼‍♂️ 

Sure thing, will do!